lib/Dialect/FIRRTL/Transforms/CreateSiFiveMetadata.cpp

```diff
@@ -246,8 +248,13 @@ struct ObjectModelIR {
 
     auto memPaths = instancePathCache.getAbsolutePaths(mem);
     SmallVector<Value> memoryHierPaths;
+    SmallVector<Value> finalInstanceNames;
     for (auto memPath : memPaths) {
-      Operation *finalInst = memPath.leaf();
+      auto finalInst = memPath.leaf();
+      finalInstanceNames.emplace_back(
+          builderOM.create<StringConstantOp>(finalInst.getInstanceNameAttr()));
+      memPath.dropBack();
+      finalInst = memPath.leaf();
```

Comment:
Is this intending to update `finalInst` to the leaf after dropping the last element? I was under the impression `dropBack()` was not side-effecting, so maybe it would need to be `finalPath = memPath.dropBack().leaf()`? Or maybe I misunderstood.

Reply:
You are right, fixed in latest version.